### PR TITLE
RAINCATCH-31: Dropdown field

### PR DESCRIPTION
**Pull request by pmdarrow**
Implement the dropdown field available in the "drag & drop apps" feature.

### CODE REVIEW

File: lib/angular/fieldDropdown/fieldDropdownDirective-spec.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+  it("The third option should be checked", function() {
+    should(this.element.find('md-option')[2].getAttribute('selected')).equal('selected');
+  });
+
+  it("The model should be set to fieldDropdownCtrl.field.value", function() {
+    should(this.element.find('md-select').attr('ng-model')).equal('fieldDropdownCtrl.field.value');
+  });
+
+  it("Setting a value on the scope should give a default value", function() {
```
Comment (pmdarrow) [inline lib/angular/fieldDropdown/fieldDropdownDirective-spec.js:88]: @nialldonnellyfh I'm having trouble getting this test to pass. I think the issue is that:

* The `fieldOptions.definition.options` for a dropdown field do not have any sort of IDs associated with them; just `label` and `checked`.
* I'm unable to use ng-model-options "track by" feature due to this. 

Shouldn't the drag & drop forms API be providing unique IDs for dropdown options?

**Comment by wtrocki** [comment]
Thanks for contribution. Changes aren't ready to merge at the moment. 
Closing as stale PR. Feel free to reopen if needed.

**Comment by wtrocki** [comment]
Reopening as this changes need to land on https://github.com/feedhenry-raincatcher/raincatcher-appform/pull/15

**Comment by wtrocki** [comment]
Fixes done as part of https://github.com/feedhenry-raincatcher/raincatcher-appform/pull/34. Closing

**Pull request closed without merging**